### PR TITLE
feat: collect slow queries in monitoring mode

### PR DESCRIPTION
**Pull request by zyy17**
## Release Notes

- **New Features**
	- Introduced a `SlowQuery` configuration for enhanced logging, allowing users to enable slow query logging with customizable thresholds and sampling ratios across various components.
	- Added documentation for the new `SlowQuery` settings in the API references.

- **Bug Fixes**
	- Improved validation for the new logging fields to ensure correct configuration.

- **Documentation**
	- Expanded API documentation to include details about the new `SlowQuery` logging capabilities.

- **Chores**
	- Updated Kubernetes manifests and CRDs to include `slowQuery` properties for consistent logging configuration across components.

### CODE REVIEW

File: apis/v1alpha1/defaulting.go [hunk 1]
```diff
@@ -151,6 +151,13 @@ func (in *GreptimeDBCluster) defaultSpec() *GreptimeDBClusterSpec {
 
 		// Set the default logging format to JSON if monitoring is enabled.
 		defaultSpec.Logging.Format = LogFormatJSON
+
+		// Turn on the slow query log by default if monitoring is enabled.
+		defaultSpec.Logging.SlowQuery = &SlowQuery{
+			Enabled:     true,
+			Threshold:   "10s",
```
Comment (daviderli614) [inline apis/v1alpha1/defaulting.go:158]: Question: Enabling this by default affect GreptimeDB performance?

File: pkg/dbconfig/common.go [hunk 2]
```diff
@@ -167,4 +178,18 @@ func (c *LoggingConfig) ConfigureLogging(spec *v1alpha1.LoggingSpec) {
...
+
+	if spec.SlowQuery != nil {
+		c.EnableSlowQuery = pointer.Bool(spec.SlowQuery.Enabled)
+		c.SlowQueryThreshold = pointer.String(spec.SlowQuery.Threshold)
+
+		// Turn string to float64
+		ration, err := strconv.ParseFloat(spec.SlowQuery.SampleRatio, 64)
+		if err != nil {
+			klog.Warningf("Failed to parse slow query sample ratio '%s', use the default value 1.0", spec.SlowQuery.SampleRatio)
+			ration = 1.0
```
Comment (daviderli614) [inline pkg/dbconfig/common.go:190]: We can set the default value of `ration = 1.0` in `defaulting.go`.

Comment (zyy17) [reply pkg/dbconfig/common.go:190]: It's already in `defaulting.go`.

File: apis/v1alpha1/defaulting.go (hunk 1 above)
Comment (zyy17) [reply apis/v1alpha1/defaulting.go:152]: Most of time, it would not affect the performance. Also, we can use `sampleRatio` to control it.

**Review by daviderli614 — APPROVED** [review]

**Pull request merged**